divi/src/uint256.cpp

```diff
@@ -71,18 +71,41 @@ base_uint<BITS>& base_uint<BITS>::operator*=(uint32_t b32)
 }
 
 template <unsigned int BITS>
-base_uint<BITS>& base_uint<BITS>::operator*=(const base_uint& b)
+bool base_uint<BITS>::MultiplyBy(const base_uint& b)
 {
+    bool overflow = false;
+    int highestNonzeroA = 0;
+    int highestNonzeroB = 0;
+
```

Comment:
We need them because the inner loop will only process terms that are not going to overflow anyway (so we might miss some overflows, even the unit test fails without those extra checks).